mne/epochs.py

```diff
@@ -3401,6 +3571,11 @@ def _concatenate_epochs(epochs_list, with_data=True, add_offset=True, *,
         if not isinstance(epochs, BaseEpochs):
             raise TypeError('epochs_list[%d] must be an instance of Epochs, '
                             'got %s' % (ei, type(epochs)))
+
+        if hasattr(epochs, 'annotations'):
```

Comment:
It only matters if they are non-None, I'm surprised this doesn't break some existing test where we create two `Epochs` instances from raw and concatenate them. Maybe we only currently test concat with EpochsArray or something? 
```suggestion
        if getattr(epochs, 'annotations', None) is not None and not warned:
            warned = True
```
And you'll need to add a `warned = False` before the loop. Basically when concatenating 10 Epochs instances with metadata you should get 1 warning, not 9.

Reply:
Well in all the concatenation tests, they might not have set annotations inside the Raw when making them into `Epochs`, so probably wasn't caught.

Reply:
Could it be that you're ignoring those warnings? I thought you added an ignore line to conftest? (Sorry, on mobile, hard to check for myself at the moment)

Reply:
Ah yeah true. So there's 23 places that `concatenate_epochs` is called unfortunately in `test_epochs.py`. I could add a catch warning statement in those files?

I suppose it depends on our timeframe or feasibility of adding concatenating Epochs with Annotations. I don't think it is possible because different Epochs might have been taken from completely different files right?

If so, I can add these catch warnings in and remove the ignore statement in `conftest.py`

Reply:
If you make my suggested change, I doubt many tests (if any) will break. They will only break off we have annotations on raw and make epochs and concat (right?), which I don't think we do

Reply:
So the tests all do break actually because there are potentially Annotations within the Raw objects. Moreover, when you do `raw.set_annotations(None)`, the following lines of code ensure that Annotations are never None again:

```
if annotations is None:
            self._annotations = Annotations([], [], [], meas_date)
```

Therefore, we were on the right track with the suggested change, but we have to further check if the `epochs.annotations` is just an empty data structure or not.

Reference:
- https://github.com/mne-tools/mne-python/blob/3d4a4f66232448de542ac076df7b57115644978a/mne/io/base.py#L675-L676